core/accounting.go

```diff
@@ -0,0 +1,72 @@
+package core
+
+import (
+	"math/big"
+	"sync"
+	"time"
+)
+
+// Use transcodeLoopTimeout as TTL for a ManifestID's credit balance
+// We should prevent the use of global state if they are not constants
+
+// Accounts holds credit balances on a per-stream basis
+type Accounts struct {
+	Balances map[ManifestID]balance
+	Mtx      sync.RWMutex
+}
+
+type balance struct {
+	LastUpdate time.Time // Unix time since last update
+	Amount     *big.Rat  // Balance represented as a big.Rat
+}
+
+// Credit adds an an amount to the balance for a ManifestID
+func (a *Accounts) Credit(id ManifestID, amount *big.Rat) {
+	a.Mtx.Lock()
+	defer a.Mtx.Unlock()
+	if a.Balances[id].Amount == nil {
+		a.Balances[id] = balance{Amount: big.NewRat(0, 1)}
+	}
+	a.Balances[id] = balance{
```

Comment:
`cannot assign to struct field in map` compiler error, regardless whether it is initialized or not. 

Only way to circumvent is use a pointer as value to a key in the map, rather than a value itself it seems? 

`balances map[ManifestID]*balance`

Reply:
Ah right the map is tracking values instead of pointers right now. I think we should track pointers since we'll be doing frequent balance updates so we should avoid creating a struct each time and instead just do an in-place update on a pointer to an existing struct.